### PR TITLE
Support push directories

### PR DESCRIPTION
**Pull request by tgalal**

### CODE REVIEW

File: adb/adb_commands.py [hunk 1]
```diff
@@ -151,6 +151,11 @@ def Push(self, source_file, device_filename, mtime='0', timeout_ms=None):
       mtime: Optional, modification time to set on the file.
       timeout_ms: Expected timeout for any part of the push.
     """
+
+    if os.path.isdir(source_file):
+      for dir in os.listdir(source_file):
+        return self.Push(os.path.join(source_file, dir), device_filename + "/" + dir)
```
Comment (fahhem) [inline adb/adb_commands.py]: You're only returning the first one. You have to Push all of them then return after the for loop

**Comment by fahhem** [comment]
Thanks for that!

**Comment by tgalal** [comment]
updated, + I'm now calling mkdir on device to include empty directories as well.

**Comment by fahhem** [comment]
LGTM

**Comment by fahhem** [comment]
Would you like to be in the CONTRIBUTORS file? If so, I'll accept either a merge request or just reply with your name and email

**Pull request merged**